packages/code/src/Code.tsx

```diff
@@ -169,16 +338,63 @@ function Code({
     </Syntax>
   );
 
+  function handleScroll(e: React.UIEvent) {
+    const scrollWidth = e.currentTarget.scrollWidth;
+    const elementWidth = e.currentTarget.clientWidth;
+    const isScrollable = scrollWidth > elementWidth;
+
+    if (isScrollable) {
+      const scrollPosition = e.currentTarget.scrollLeft;
+      const maxPosition = scrollWidth - elementWidth;
+
+      if (scrollPosition > 0 && scrollPosition < maxPosition) {
+        dispatch({ type: 'scrollBoth' });
+      } else if (scrollPosition > 0) {
+        dispatch({ type: 'scrollLeft' });
+      } else if (scrollPosition < maxPosition) {
+        dispatch({ type: 'scrollRight' });
+      }
+    }
+  }
+
   if (!multiline) {
     return (
       <div className={wrapperStyle}>
         {renderedWindowChrome}
 
         <div
-          {...(rest as DetailedElementProps<HTMLDivElement>)}
-          className={wrapperClassName}
+          className={css`
+            display: flex;
+          `}
         >
-          {renderedSyntaxComponent}
+          <div
+            {...(rest as DetailedElementProps<HTMLDivElement>)}
```

Comment:
Usually we prefer to spread `rest` on the root element. Any particular reason you did it here?

Reply:
This is where it was spread initially, so I can't really speak to the reasoning.

Reply:
Hmm... might need to explore this a bit more after this change then.

Reply:
I think the idea is that `rest` is spread on the `<pre>` or `<div>` element that actually holds  `renderedSyntaxComponent` so that you can make changes to how the code is displayed... but that's just my guess.